File: backend/SportsCenterBackend/SportsCenter.Application/Clients/Queries/GetClientsByAge/GetClientsByAge.cs
using SportsCenter.Application.Abstractions;

namespace SportsCenter.Application.Clients.Queries.GetClientsByAge;

public class GetClientsByAge : IQuery<IEnumerable<ClientByAgeDto>>
{
    public int MinAge { get; set; }
    public int MaxAge { get; set; }
    public int Offset { get; set; }

    public GetClientsByAge(int offSet, int minAge, int maxAge)
    {
        Offset = offSet;
        MinAge = minAge;
        MaxAge = maxAge;
    }
}

File: frontend/src/api/getClientsByAge.js
import API_URL from "../appConfig";

export default async function getClientsByAge(token, minAge, maxAge, offset) {
    return fetch(`${API_URL}/Clients/byAge/?minAge=${Number(minAge)}&maxAge=${Number(maxAge)}&offset=${offset}`, {
        method: 'GET',
        headers: {
            'Content-Type': 'application/json'
        },
        credentials: 'include',
    });
}
